mars/dataframe/merge/merge.py

```diff
@@ -113,16 +115,24 @@ def execute_map(cls, ctx, op):
         for index_idx, index_filter in enumerate(filters):
             reducer_index = (index_idx, chunk.index[1])
             if index_filter is not None and index_filter is not list():
-                ctx[chunk.key, reducer_index] = df.iloc[index_filter]
+                ctx[chunk.key, reducer_index] = (
+                    op.mapper_id,
+                    ctx[op].index,
```

Comment:
Key of `ctx` are basically keys and tuples, why trying to get `op` from ctx?

Reply:
Some operands such as `align`/`merge`/`unique` rely on mapper chunk index/mapperid for reduce execution, so we need a way to write mapper index/mapperid into mapper result. The api `Operand#execute(ctx, op)` doesn't has chunk as input parameters, and it will incur a lots of change if we change `Operand#execute(ctx, op)` to `Operand#execute(ctx, chunk)`

Reply:
Added `get_current_chunk`/`set_current_chunk` API instead.